File: Server/FIFA.Server/Models/Match/ResultViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FIFA.Server.Models
{

    // Class representing a team player with the number of goals
    public class TeamPlayerResultViewModel
    {
        public int Id { get; set; }
        public string TeamName { get; set; }
        public string PlayerName { get; set; }
        public int nbGoals { get; set; }
    }

    // Class representing a match result
    public class MatchResultViewModel
    {
        public int LeagueId { get; set; }
        public string LeagueName { get; set; }
        public int SeasonId { get; set; }
        public string SeasonName { get; set; }
        public int CountryId { get; set; }
        public string CountryName { get; set; }
        public DateTime? Date { get; set; }
        public TeamPlayerResultViewModel homeTeamPlayer { get; set; }
        public TeamPlayerResultViewModel awayTeamPlayerName { get; set; }
    }

    // Class representing a league with the played matches for a date
    public class LeagueResultViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime? Date { get; set; }

        public IEnumerable<MatchResultViewModel> matches { get; set; }
    }

    // Class representing a season with the played matches in a league for a date
    public class SeasonResultViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime? Date { get; set; }

        public IEnumerable<LeagueResultViewModel> leagueMatches { get; set; }

    }

    // Class representing a country with the played matches in a season / league for a date
    public class CountryResultViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime? Date { get; set; }

        public IEnumerable<SeasonResultViewModel> seasonMatches { get; set; }

    }



    // Class representing a match with it s date and the players
    public class ResultViewModel
    {
        public DateTime? Date { get; set; }

        public IEnumerable<CountryResultViewModel> countryMatches { get; set; }
    }


}

File: Client/typescript/module/results/ResultViewModel.ts
module FifaLeagueClient.Module.Results {

	export class ResultViewModel {

      public Date: Date;
	    public countryMatches: CountryResultViewModel[];

			// build the model directly from the data returned by the service
			constructor(data){
					if(data != null){
							this.Date = data.Date;
			        this.countryMatches = data.countryMatches;
					}
			}
	}

export class CountryResultViewModel {
	public Id: number;
	public Name: string;
	public Date: Date;

	public seasonMatches: SeasonResultViewModel[];

	contructor(data){
		if (data != null){
			this.Id = data.Id;
			this.Name = data.Name;
			this.Date = data.Date;
			this.seasonMatches = data.seasonMatches;
		}
	}
}

	export class SeasonResultViewModel {
		public Id: number;
		public Name: string;
		public Date: Date;

		public leagueMatches: LeagueResultViewModel[];

		contructor(data){
			if (data != null){
				this.Id = data.Id;
				this.Name = data.Name;
				this.Date = data.Date;
				this.leagueMatches = data.leagueMatches;
			}
		}
	}

  export class LeagueResultViewModel {
    public Id: number;
    public Name: string;
		public Date: Date;

		public matches: MatchResultViewModel[];

    contructor(data){
      if (data != null){
        this.Id = data.Id;
        this.Name = data.Name;
				this.Date = data.Date;
				this.matches = data.matches;
      }
    }
  }

	export class MatchResultViewModel {
		public Id:number;
		public leagueId: number;
		public leagueName: string;
		public Date: Date;

		public homeTeamPlayer: TeamPlayerResultViewModel;
		public awayTeamPlayer: TeamPlayerResultViewModel;


		// build the model directly from the data returned by the service
		constructor(data){
				if(data != null){
					this.Id = data.Id;
					this.leagueId = data.leagueId;
					this.leagueName = data.leagueName;
					if(data.Date != null){
						this.Date = new Date(data.Date);
					}
					this.homeTeamPlayer = data.homeTeamPlayer;
					this.awayTeamPlayer = data.awayTeamPlayer;
				}
		}


	}

	export class TeamPlayerResultViewModel {
		public Id: number;
		public TeamName: string;
		public PlayerName: string;
		public nbGoals: number;

		contructor(data){
			if (data != null){
				this.Id = data.Id;
				this.TeamName = data.TeamName;
				this.PlayerName = data.PlayerName;
				this.nbGoals = data.nbGoals;
			}
		}
	}


}
